### PR TITLE
General cleanup of schema definition files

### PR DESCRIPTION
**Pull request by floydtree**
#### Related Issue: 
Partially related - https://github.com/ocsf/ocsf-server/issues/88

#### Description of changes:
1. Adding requirement attributes in schema definition files, wherever missing. Making requirement definition consistent throughout the framework. This will eliminate warnings during server builds.
2. Minor bugfixes in schema definition files
3. Adding a basic gitignore
4. Adjusting the pr template to remove requirement of having labels. (only github org members are able to assign labels so non-members aren't able to assign them. We can let PR reviewers can assign labels as necessary.

### CODE REVIEW

**Comment by mikeradka** [comment]
This is great. Question for you @floydtree - I noticed yesterday that the dictionary attributes aren't wholly alphabetized and was considering putting in a PR. Is this something you'd like to include in this cleanup, or would you rather I put in a fresh PR?

**Comment by floydtree** [comment]
> This is great. Question for you @floydtree - I noticed yesterday that the dictionary attributes aren't wholly alphabetized and was considering putting in a PR. Is this something you'd like to include in this cleanup, or would you rather I put in a fresh PR?

Yep good catch! I'll include it in this PR. The change should now be in.

**Review by zschmerber — APPROVED** [review]
Thanks for all of the cleanup.

**Review by mikeradka — APPROVED** [review]
LGTM

File: events/discovery/device_config_state_change.json [hunk 1]
```diff
@@ -1,41 +1,47 @@
...
+  "uid": 19,
   "caption": "Device Config State Change",
   "description": "Device Config State Change events report state changes that impact the security of the device.",
   "extends": "discovery",
   "name": "device_config_state_change",
-  "uid": 19,
-  "profiles": [
-    "host"
-  ],
   "attributes": {
```
Comment (pagbabian-splunk) [inline events/discovery/device_config_state_change.json:7]: Why did we remove the profile declaration for host?  Seems like the object natively builds-in host so it should be declared.

Comment (floydtree) [reply events/discovery/device_config_state_change.json:7]: It's not removed, the json was sorted alpha numerically.

**Review by pagbabian-splunk — APPROVED** [review]

**Pull request merged**